### DISCUSSION TITLE
Foundry Error after installation: `GLIBC_2.29' not found

### DISCUSSION THREAD

**Discussion by Play3rZer0** [Cyfrin/foundry-full-course-cu#2443] [General]
After installing Foundry, I tried to verify the forge version. Here is the error I get:

$ forge --version
forge: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by forge)
forge: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.29' not found (required by forge)    
forge: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by forge) 

I tried reinstalling and after foundryup, the error persists. 

I am running this on a Windows 10 system with Ubuntu (WSL) 18.04.6 LTS. I followed the procedure from the getfoundry.sh website. It appears to install, but mentions the following at the end of the installation process:

```
foundryup: command failed: /home/vtech/.foundry/bin/chisel --version
foundryup: installed -
foundryup: done!
```
If anyone has encountered similar or same problem, what was your solution (if any)?

**Comment by EngrPips** [comment]
Hello, please read this discussion which is from the official foundry repo https://github.com/foundry-rs/foundry/issues/3827 can you try to update Ubuntu and retry?

`
FYI:
I have just faced the same error locally when cloning and installing repo. Once I upgraded from Ubuntu 20.04 to 22.04 (on WSL) everything is back to normal.
`

**Reply by Play3rZer0** [reply]
Problem solved. Here is what I did:
1. Upgrade Ubuntu by going to the Microsoft Store to get Ubuntu 20.04.6 LTS.
2. Install Ubuntu 20.04.6 LTS
3. Open VS Code and create new terminal using the Ubuntu shell
4. Run the installation script for Foundry
5. Run foundryup
6. Check if properly running by entering forge --version
7. Get response -> forge 0.2.0 (be451fb 2024-09-11T00:20:35.367236373Z)

I tried other solutions but did not work. One suggested solution I saw on Stackoverflow was to download the GLIB library, update the version by recompiling the os and running several scripts to point to the desired path. It did not work for me. Simplest solution is to upgrade from Ubuntu 18 to 20. I am not sure about the latest version of Ubuntu, but Ubuntu 20.04.6 LTS works fine.

**Reply by EngrPips** [reply]
Thanks for the heads-up sir.
